dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Basics/SwitchOff/SwitchOffLogic.mo

```diff
@@ -14,15 +14,20 @@ within Dynawo.Electrical.Controls.Basics.SwitchOff;
 */
 
 partial model SwitchOffLogic "Manage switch-off logic"
-  /* Handles a predefinite number of switch-off signals and sets running to false as soon as one signal is set to true */
 
-  Dynawo.Connectors.BPin switchOffSignal1(value(start = false)) "Switch-off signal 1";
-  Dynawo.Connectors.BPin switchOffSignal2(value(start = false)) if NbSwitchOffSignals >= 2 "Switch-off signal 2";
-  Dynawo.Connectors.BPin switchOffSignal3(value(start = false)) if NbSwitchOffSignals >= 3 "Switch-off signal 3";
+  parameter Integer NbSwitchOffSignals(min = 1, max = 3) "Number of switch-off signals to take into account in inputs";
 
-  Dynawo.Connectors.BPin running(value(start = true)) "Indicates if the component is running or not";
+  Dynawo.Connectors.BPin switchOffSignal1(value(start = SwitchOffSignal10)) "Switch-off signal 1";
+  Dynawo.Connectors.BPin switchOffSignal2(value(start = SwitchOffSignal20)) if NbSwitchOffSignals >= 2 "Switch-off signal 2";
+  Dynawo.Connectors.BPin switchOffSignal3(value(start = SwitchOffSignal30)) if NbSwitchOffSignals >= 3 "Switch-off signal 3";
 
-  parameter Integer NbSwitchOffSignals(min = 1, max = 3) "Number of switch-off signals to take into account in inputs";
+  Dynawo.Connectors.BPin running(value(start = Running0)) "Indicates if the component is running or not";
+
+  parameter Boolean SwitchOffSignal10 = false "Initial switch-off signal 1";
+  parameter Boolean SwitchOffSignal20 = false if NbSwitchOffSignals >= 2 "Initial switch-off signal 2";
+  parameter Boolean SwitchOffSignal30 = false if NbSwitchOffSignals >= 3 "Initial switch-off signal 3";
```

Comment:
I'm not sure if we need the "if" here. 

Did you verify that all the OpenModelica examples work?

Reply:
If NbSwitchOffSignals = 2, the parameter SwitchOffSignal30 does not exist (see the dumpInit.txt), like the connector switchOffSignal3. This is consistent.

Reply:
All OpenModelica examples are well simulated, except two whose simulation terminates early even on master branch : GridForming, WPP4ACurrentSource. 

See issue [#2977 ](https://github.com/dynawo/dynawo/issues/2977)

Reply:
Yes, I totally understand the reason to put the if.

But since I have another version of OM, I get an error for all the examples where all of them are using the switch-off signal.
The error is "[Component Running0 of variability parameter has binding 'if gridImpedance.NbSwitchOffSignals >= 3 then (not gridImpedance.SwitchOffSignal10 and not gridImpedance.SwitchOffSignal20) and not gridImpedance.SwitchOffSignal30 else if gridImpedance.NbSwitchOffSignals >= 2 then not gridImpedance.SwitchOffSignal10 and not gridImpedance.SwitchOffSignal20 else not gridImpedance.SwitchOffSignal10' of higher variability continuous.]"

I thought the problem came from the fact that Running0 is a final parameter, but even without the "final" keyword I get the same error. By keeping one if condition: the one with Running0, the error is solved, but I am not totally convinced.